### PR TITLE
Fix/use correct package manager

### PR DESCRIPTION
**Pull request by stefan-karlsson**
Details: 
The code before did not detect which package manager that should be used.
Instead it just checked if 'yarn' was allowed to use. 

This code will first check if `package-lock.json` or `yarn.lock` file exists. If that is not possible the package `detect-package-manager` will try to execute the `npm` and `yarn` commands to determine if it is installed. 

**NOTE** Having some issues running the `yarn affected:e2e` command locally so i am not able to test this solution out from a users perspective. Would be nice with some help here.

Another thing is that we really should write some tests for the implementation of the package manager. Right now i am not sure if `NPM.ts` and `Yarn.ts` works as expected.

Breaking Changes:
Well i guess there is a breaking change since it will work differently. But since the code already was breaking i am not sure if this is better or worse 👍 

Fixes/Feature #50 

- [x] yarn affected:test succeeds
- [ ] yarn affected:e2e succeeds
- [x] yarn format:check succeeds
- [x] Code coverage does not decrease (if any source code was changed)
- [ ] If applicable, appropriate Wiki docs were updated (if applicable)
- [ ] If applicable, code review comments are written in the source code with / {Name}

### CODE REVIEW

**Comment by wickstargazer** [comment]
I will review it tonight and get it going 😄

File: libs/nx-serverless/src/utils/packagers/index.ts [hunk 1]
```diff
@@ -64,121 +26,127 @@ export function preparePackageJson(
...
   resolverName: string,
   tsconfig?: string
-): Observable<BuilderOutput> {
+): Promise<BuilderOutput> {
+  const packageJsonPath = join(options.package, 'package.json');
+  const cwd = dirname(packageJsonPath);
   const resolver = resolverFactory(resolverName, context);
   context.logger.info('getting external modules');
   const workspacePackageJsonPath = join(context.workspaceRoot, 'package.json');
-  const packageJsonPath = join(options.package, 'package.json');
```
Comment (wickstargazer) [inline libs/nx-serverless/src/utils/packagers/index.ts:71]: why was this changed? The purpose is to get the minimal `package.json`

Comment (stefan-karlsson) [reply libs/nx-serverless/src/utils/packagers/index.ts:71]: It has not been changed, if you check line 30 its still there. If you ask why i moved it, i dont have a good reason :)

**Comment by wickstargazer** [comment]
i am closing this pull request since we copied over some of ure stuff and made 0.5.3
thankz for the effort really showed us the way!
